src/auth/models.cpp

```diff
@@ -464,6 +483,78 @@ bool operator==(const Role &first, const Role &second) {
   return first.rolename_ == second.rolename_ && first.permissions_ == second.permissions_;
 }
 
+#ifdef MG_ENTERPRISE
+void Databases::Add(const std::string &db) {
+  if (allow_all_) {
+    grants_dbs_.clear();
+    allow_all_ = false;
+  }
+  grants_dbs_.emplace(db);
+  denies_dbs_.erase(db);
+}
+
+void Databases::Remove(const std::string &db) {
+  denies_dbs_.emplace(db);
+  grants_dbs_.erase(db);
+}
+
+void Databases::Delete(const std::string &db) {
+  denies_dbs_.erase(db);
+  if (!allow_all_) {
+    grants_dbs_.erase(db);
+  }
```

Comment:
Yeah, I think this logic needs to be updated.
Currently I default to allow_all(true) and default_db_(kDefaultDB). But a better solution would be to default to false and none.
Then it would be up to the admin to define exactly what they want.

Reply:
This will be a bit different in the end. 
In order to have the same behavior as before, the default db will be set to kDefaultDB, but allow_all will be false and the only databases granted will be the kDefaultDB. 
The idea is to have everything work the with and without the multi-tenancy. I also default to this when a user saved with the previous version is read.